bin/pycbc_multi_inspiral

```diff
@@ -177,112 +366,229 @@ with ctx:
-
-                snr, norm, corr, idx, snrv = \
-                        matched_filter.matched_filter_and_cluster(s_num,
-                            template.sigmasq(stilde.psd), cluster_window)
-
-                if len(idx) == 0:
-                    continue
-
-                out_vals['bank_chisq'], out_vals['bank_chisq_dof'] = \
-                        bank_chisq.values(template, stilde.psd, stilde, snrv,
-                                          norm, idx+stilde.analyze.start)
-
-                out_vals['chisq'], out_vals['chisq_dof'] =\
-                        power_chisq.values(corr, snrv, norm, stilde.psd,
-                                           idx+stilde.analyze.start, template)
-
-                out_vals['cont_chisq'] = autochisq.values(snr,
-                        idx+stilde.analyze.start, template, stilde.psd, norm,
-                        stilde=stilde, low_frequency_cutoff=flow)
-
-                idx += stilde.cumulative_index
-
-                out_vals['time_index'] = idx
-                out_vals['snr'] = snrv * norm
-                # IFO is stored as an int
-                out_vals['ifo'] = event_mgr.ifo_dict[ifo]
-
-                event_mgr.add_template_events_to_ifo(ifo, names,
-                        [out_vals[n] for n in names])
-
-            event_mgr.cluster_template_events_single_ifo("time_index",
-                                                    "snr", cluster_window, ifo)
+			       low_frequency_cutoff=flow, phase_order=opt.order,
+			       taper=opt.taper_template,
+			       approximant=opt.approximant, out=template_mem)
+    Fp = {} #Antenna patterns
+    Fc = {}
+    for ifo in opt.instruments:
+        Fp[ifo], Fc[ifo] = pycbc.detector.Detector(ifo).antenna_pattern(\
+                                                     ra, dec, pol, t_gps)
+
+    for t_num, template in enumerate(bank): #Loop over templates
+	for ifo in opt.instruments:
+	    if opt.cluster_method == "window":
+		cluster_window = int(opt.cluster_window * sample_rate)
+	    elif opt.cluster_method == "template":
+		cluster_window = int(template.chirp_length * sample_rate)
+	    elif opt.cluster_window == 0:
+		 cluster_window = int(0)
+
+	for s_num,stilde in enumerate(segments[opt.instruments[0]]): #Loop over segments
+            logging.info("Analyzing segment %d/%d" % (s_num + 1, len(segments[ifo]))) 
+            #Find detector sensitivities (sigma) and make array of normalised 
+            sigmasq = {ifo : template.sigmasq(segments[ifo][s_num].psd) for ifo in opt.instruments}
+            sigma = {ifo : np.sqrt(sigmasq[ifo]) for ifo in opt.instruments}
+            stilde = {ifo : segments[ifo][s_num] for ifo in opt.instruments}
+            #Every time s_num is zero, we run new template to increment the template index
+            if s_num==0: event_mgr.new_template(tmplt=template.params, sigmasq=sigmasq)
+            snrv_dict = {}
+            norm_dict = {}
+            corr_dict = {}
```

Comment:
Indentation seems strange here??

Reply:
Why? 
This is the first time I've ever really had to care about how my code looks but I can't see anything strange here.